provider/telegram.go

```diff
@@ -0,0 +1,142 @@
+package provider
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/abahmed/kwatch/event"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	telegramAPIURL = "https://api.telegram.org/bot%s/sendMessage"
+)
+
+type telegram struct {
+	token  string
+	chatId string
+}
+
+// NewTelegram returns a new Telegram object
+func NewTelegram(token string, chatId string) Provider {
+	// object validation
+	if len(token) == 0 {
+		logrus.Warnf("initializing telegram with empty token")
+	} else if len(chatId) == 0 {
+		logrus.Warnf("initializing telegram with empty chat_id")
+	} else {
+		logrus.Infof("initializing telegram with token  %s and chat_id %s", token, chatId)
+	}
+
+	// returns a new telegram object
+	return &telegram{
+		token:  token,
+		chatId: chatId,
+	}
+}
+
+// Name returns name of the provider
+func (t *telegram) Name() string {
+	return "Telegram"
+}
+
+// SendEvent sends event to the provider
+func (t *telegram) SendEvent(e *event.Event) error {
+	logrus.Debugf("sending to telegram event: %v", e)
+
+	// validate telegram token and chat Id
+	err, _ := validateTelegram(t)
+	if err != nil {
+		return err
+	}
+
+	return sendByTelegramApi("", t, e)
+}
```

Comment:
@Andrew44Ashraf what do you think about moving the `buildRequestBodyTelegram` call to here, passing it as the first string parameter, and removing the event parameter from `sendByTelegramApi`

Reply:
@amgadramses good one, refactored the code please check.